src/main/java/com/rikaisan/mixin/BlockLogicWireRedstoneMixin.java

```diff
@@ -9,63 +14,31 @@
-				|| !worldSource.isBlockNormalCube(x + 1, y, z) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x + 1, y - 1, z, -1, Side.TOP);
-			boolean negZShouldConnectTo = shouldConnectTo(worldSource, x, y, z - 1, 2)
-				|| !worldSource.isBlockNormalCube(x, y, z - 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y - 1, z - 1, -1, Side.TOP);
-			boolean posZShouldConnectTo = shouldConnectTo(worldSource, x, y, z + 1, 0)
-				|| !worldSource.isBlockNormalCube(x, y, z + 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y - 1, z + 1, -1, Side.TOP);
-
-			if (!worldSource.isBlockNormalCube(x, y + 1, z)) {
-				if (worldSource.isBlockNormalCube(x - 1, y, z) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x - 1, y + 1, z, -1, Side.BOTTOM)) {
-					negXShouldConnectTo = true;
-				}
 
-				if (worldSource.isBlockNormalCube(x + 1, y, z) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x + 1, y + 1, z, -1, Side.BOTTOM)) {
-					posXShouldConnectTo = true;
-				}
 
-				if (worldSource.isBlockNormalCube(x, y, z - 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y + 1, z - 1, -1, Side.BOTTOM)) {
-					negZShouldConnectTo = true;
-				}
-
-				if (worldSource.isBlockNormalCube(x, y, z + 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y + 1, z + 1, -1, Side.BOTTOM)) {
-					posZShouldConnectTo = true;
-				}
-			}
+	@WrapOperation(method = "getSignal", at = @At(value = "INVOKE", target = "Lnet/minecraft/core/block/BlockLogicWireRedstone;shouldConnectTo(Lnet/minecraft/core/world/WorldSource;IIII)Z"))
+	private boolean fixDiagonal(WorldSource worldSource, int x, int y, int z, int data, Operation<Boolean> original, @Local(name = "y") int originalY) {
+		if(y == originalY - 1) return AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y, z, data, Side.TOP);
+		if(y == originalY + 1) return AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y, z, data, Side.BOTTOM);
+		return original.call(worldSource, x, y, z, data);
+	}
 
+	@Definition(id = "negZShouldConnectTo", local = @Local(type = boolean.class, name = "negZShouldConnectTo"))
+	@Expression("negZShouldConnectTo")
+	@Inject(method = "getSignal", at = @At("MIXINEXTRAS:EXPRESSION"), cancellable = true)
+	private void fixRedirections(WorldSource worldSource, int x, int y, int z, Side side, CallbackInfoReturnable<Boolean> cir, @Local(name = "posXShouldConnectTo") boolean posXShouldConnectTo, @Local(name = "negXShouldConnectTo") boolean negXShouldConnectTo, @Local(name = "posZShouldConnectTo") boolean posZShouldConnectTo, @Local(name = "negZShouldConnectTo") boolean negZShouldConnectTo) {
 		boolean isXConnected = posXShouldConnectTo || negXShouldConnectTo;
 		boolean isZConnected = posZShouldConnectTo || negZShouldConnectTo;
 
 		// Refactor + Patch to make redirection work as intended
```

Comment:
Comment should be moved above the inject, as this method is now purely about this fix and refactor.

Reply:
Fixed by 9f57f3c